apps/spa/src/pages/Connect4Page.tsx

```diff
@@ -1,50 +1,112 @@
-import React, { useReducer } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import GameGrid from "@components/GameGrid";
 import { useSearchParams } from "react-router-dom";
-import { playToken } from "../../../../packages/shared/lib/connect4";
-import useInitGameState from "@hooks/useInitGameState";
-import type { GameState } from "../../../../packages/shared/types/gameState";
+import {
+  PlayerNum,
+  type Game,
+} from "../../../../packages/shared/types/gameState";
+import ShareButton from "@components/ShareButton";
+import { getGame, getGameStateVersion } from "@services/api";
+import GameInfos from "@components/GameInfos";
 
-export const BOARD_STATE_QP = "state";
-export const BOARD_STATE_QP_SEPARATOR_CHAR = ",";
+export const PLAYER_NUM_QP = "playerNum";
+export const GAME_ID_QP = "gameId";
+export const SHARED_GAME_ID_QP = "sharedGameId";
+
+async function isGameStateNeedsUpdate(game: Game): Promise<boolean> {
+  try {
+    const { stateVersion: newStateVersion } = await getGameStateVersion(
+      game.id,
+    );
+    return game.stateVersion !== newStateVersion;
+  } catch (error) {
+    console.error("Error checking state version:", error);
+    return false;
+  }
+}
 
 const Connect4Page: React.FC = () => {
```

Comment:
One thing is not clear by reading the code. The game should have 2 display modes: one when I'm waiting for my component, and I can't click anywhere. The other when I'm playing, and the click handlers are enabled. This should be front and foremost in this component (or in GameGrid)  but I don't see it.